apis/applicationsets/v1alpha1/types.go

```diff
@@ -36,6 +36,10 @@ type ApplicationSetParameters struct {
 	ApplyNestedSelectors         bool                            `json:"applyNestedSelectors,omitempty" protobuf:"bytes,8,name=applyNestedSelectors"`
 	IgnoreApplicationDifferences ApplicationSetIgnoreDifferences `json:"ignoreApplicationDifferences,omitempty" protobuf:"bytes,9,name=ignoreApplicationDifferences"`
 	TemplatePatch                *string                         `json:"templatePatch,omitempty" protobuf:"bytes,10,name=templatePatch"`
+
+	// AppsetNamespace is the namespace of the application set in the ArgoCD server
+	// +optional
+	AppsetNamespace string `json:"appsetNamespace"`
```

Comment:
Optional fields should be pointers and have the `omitempty` json tag. The `+optional` is not necessary then because it is implied.

```suggestion
	AppsetNamespace *string `json:"appsetNamespace,omitempty"`
```

Reply:
Fixed